src/appleseed/renderer/modeling/aov/diffuseaov.cpp

```diff
@@ -58,22 +58,31 @@ namespace
     //
 
     class DiffuseAOVAccumulator
-      : public ColorAOVAccumulator
+      : public AOVAccumulator
     {
       public:
         explicit DiffuseAOVAccumulator(const size_t index)
-          : ColorAOVAccumulator(index)
+          : m_index(index)
         {
         }
 
         virtual void write(
+            const PixelContext&         pixel_context,
+            const ShadingPoint&         shading_point,
             const ShadingComponents&    shading_components,
-            const float                 multiplier) override
+            ShadingResult&              shading_result) override
         {
-            m_color = shading_components.m_diffuse.to_rgb(g_std_lighting_conditions);
-            m_color += shading_components.m_indirect_diffuse.to_rgb(g_std_lighting_conditions);
-            m_color *= multiplier;
+            shading_result.m_aovs[m_index].rgb() =
+                shading_components.m_diffuse.to_rgb(g_std_lighting_conditions);
+
+            shading_result.m_aovs[m_index].rgb() +=
```

Comment:
For readability and performance, collapse these two lines?

    shading_result.m_aovs[m_index].rgb() =
        shading_components.m_diffuse.to_rgb(g_std_lighting_conditions) +
        shading_components.m_indirect_diffuse.to_rgb(g_std_lighting_conditions);